OPA.ps1

```diff
@@ -13,50 +13,111 @@ $DebugPreference = "Continue"
+
+    try {
+        $WebClient = New-Object System.Net.WebClient
+        $WebClient.DownloadFile($InstallUrl, $OutFile)
+        Write-Information -MessageData "Installed the latest acceptable version of ${OPAExe} version ${ExpectedVersion}." | Out-Host
+    }
+    catch {
+        $Error[0] | Format-List -Property * -Force | Out-Host
+        Write-Error "Unable to download OPA executable. To try manually downloading, see details in README under 'Download the required OPA executable'" | Out-Host
+    }
+    finally {
+        $WebClient.Dispose()
+    }
 }
-catch {
-    Write-Error "An error has occurred: Unable to download OPA executable. To try manually downloading, see details in README under 'Download the required OPA executable'"
+
+
+function Confirm-OPAExeIsValid {
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [Alias('opa')]
+        [string]$OPAExe,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [Alias('version')]
+        [string]$ExpectedVersion
+    )
+
+    # Hash checks
+    $ExpectedHash ="5D71028FED935DC98B9D69369D42D2C03CE84A7720D61ED777E10AAE7528F399"
+    $Output = ""
+    if ((Get-FileHash .\opa_windows_amd64.exe).Hash -ne $ExpectedHash) {
+        Write-Information "SHA256 verification failed, retry download or install manually. See README under 'Download the required OPA executable' for instructions." | Out-Host
+        return $false
+    }
+
+    # Version checks
+    Try {
+        $OPAArgs = @('version')
+        $InstalledVersion= $(& "./$($OPAExe)" @OPAArgs) | Select-Object -First 1
+        if ($InstalledVersion -ne "Version: $($ExpectedVersion)") {
+            Write-Information "`Downloaded OPA version` `"$InstalledVersion`" does not meet the ScubaGear requirement of` `"$ExpectedVersion`"" | Out-Host
+            return $false
+        }
+    }
+    catch {
+        $Error[0] | Format-List -Property * -Force | Out-Host
+        Write-Error "See details on manual installation in the README under 'Download the required OPA executable'" | Out-Host
+        return $false
+    }
+
+    $Output += "Downloaded OPA version `"$InstalledVersion`" SHA256 verified successfully`n"
+    return $true, $Output
 }
-finally {
-    $WebClient.Dispose()
+
+function Install-OPA {
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [Alias('opa')]
+        [string]$OPAExe,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [Alias('version')]
+        [string]$ExpectedVersion
+    )
+    Get-OPAExecutable -opa $OPAExe -version $ExpectedVersion
+    $Result = Confirm-OPAExeIsValid -opa $OPAExe -version $ExpectedVersion
+    $Result[1] | Out-Host
 }
 
-# Hash checks
-if ((Get-FileHash .\opa_windows_amd64.exe).Hash -eq $ExpectedHash)
-    {
-    Write-Information "SHA256 verified successfully"
+# Set expected version and OutFile path
+$ExpectedVersion = "0.42.1"
+$OPAExe = "opa_windows_amd64.exe"
```

Comment:
Consider refactoring these to input parameters. Maybe using these as default values.  But pass values from Setup.ps1 so this is generic (i,e., no version or hash info). This will set us up to be able to automatically load any version in future - and automatically test against that version.

Reply:
@schrolla do you know where that script is?

Reply:
I pushed some changes, let me know if this is what you were thinking

Reply:
It aligns with my thinking.